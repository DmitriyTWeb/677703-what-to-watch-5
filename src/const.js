export const RatingDesc = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const DEFAULT_RENDERED_FILMS_COUNT = 8;
export const PLAYING_TIMEOUT = 1000;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  MYLIST: `/mylist`,
  FILMS_ID: `/films/:id`,
  PLAYER_ID: `/player/:id`,
  REVIEW_ID: `/films/:id/review`,
  ERROR: `/error`,
};

export const APIRoute = {
  FILMS: `/films`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  PROMO: `/films/promo`,
};
