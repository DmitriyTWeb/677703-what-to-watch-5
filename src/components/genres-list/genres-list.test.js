import React from "react";
import renderer from "react-test-renderer";
import GenresList from "./genres-list";
import {mockGenres} from "../../const.js";

const noop = () => { };

it(`Should GenresList render correctly`, () => {
  const tree = renderer.create(
      <GenresList
        genres={mockGenres}
        activeGenre={mockGenres[0]}
        onGenreChange={noop}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

