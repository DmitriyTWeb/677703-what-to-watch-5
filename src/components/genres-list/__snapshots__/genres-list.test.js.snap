// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenresList render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      data-genre="All genres"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Comedy"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Adventure"
      href="#"
      onClick={[Function]}
    >
      Adventure
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Triller"
      href="#"
      onClick={[Function]}
    >
      Triller
    </a>
  </li>
</ul>
`;
